benchmarks/Microsoft.AspNetCore.Routing.Performance/Matching/JumpTableSingleEntryBenchmark.cs

```diff
@@ -67,12 +86,234 @@ public int Implementation()
             var segments = _segments;
 
             var destination = 0;
-            for (var i = 0; i < strings.Length; i++)
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _implementation.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int Prototype()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _prototype.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int Trie()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
             {
-                destination = _table.GetDestination(strings[i], segments[i]);
+                destination = _trie.GetDestination(strings[i], segments[i]);
             }
 
             return destination;
         }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int VectorTrie()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _vectorTrie.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        private class SingleEntryAsciiVectorizedJumpTable : JumpTable
```

Comment:
So I plan to do another PR to merge this in. Probably subsuming the ASCII work. This is about half the cost of a string.Compare for a since string. I wanted to include it here for reference because this is basically the algorithm used in the generated code but that has more ifs.